File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
private:
    std::stack<long long> stack;
public:
    int evalRPN(vector<string>& tokens) {
        for(auto &item : tokens){
            if(isdigit(item.back())){
                stack.push(stoll(item));
            }
            else{
                long long num1, num2;
                num2 = stack.top();
                stack.pop();
                num1 = stack.top();
                stack.pop();
                if(item.back() == '+'){
                    stack.push(num1 + num2);
                }
                else if(item.back() == '-'){
                    stack.push(num1 - num2);
                }
                else if(item.back() == '/'){
                    stack.push(num1 / num2);
                }
                else{
                    stack.push(num1 * num2);
                }
            }
        }
        return stack.top();
    }
    ~Solution(){
        while(!stack.empty()) stack.pop();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for item in tokens:
            if item in ["+", "-", "*", "/"]:
                num2 = stack.pop()
                num1 = stack.pop()
                res = 0
                if item == "+":
                    res = num1 + num2
                elif item == "-":
                    res = num1 - num2
                elif item == "*":
                    res = num1 * num2
                else:
                    res = int(num1 / num2)
                stack.append(res)
            else:
                stack.append(int(item))
        return stack.pop()